stl/inc/mutex

```diff
@@ -299,22 +299,56 @@ void _Unlock_locks(const int _First, const int _Last, index_sequence<_Indices...
     (void) _Ignored;
 }
 
+template <class _Fn>
+struct _Unlock_functor_guard {
+    static_assert(is_trivially_copyable_v<_Fn>, "This scope guard is only used for trivially copyable functors.");

```

Comment:
```suggestion
    static_assert(is_trivially_copyable_v<_Fn>, "This scope guard is only used for trivially copyable function objects.");
```

personal pet peeve - functor is already an insanely overloaded term, we should not additionally add the meaning of "function object".

Additionally, below, `_Functor` -> `_Fx` or `_Func`; `_Ftor` -> `_Func`.

Reply:
I also don't really understand why the function object being trivially copyable is important.

Reply:
I suspect that "trivially copyable" is being used here because it implies `is_nothrow_copy_constructible_v`. Absent that requirement, we'd need a `try` block in the constructor to ensure the function object is called if copying it throws.

Reply:
I wanted to make a stronger assertion that initialization of the guard is not only non-throwing, but also has no side effects...